snakemake/executors/slurm/slurm_submit.py

```diff
@@ -0,0 +1,349 @@
+        restart_times=0,
+        exec_job=None,
+        max_status_checks_per_second=1,
+        cluster_config=None,
+    ):
+        # needs to be set in either case for the submission string
+        if not cores:
+            cores = 1
+
+        # seems otherswise not in namespace, despite the super class init
+        self.max_status_checks_per_second = max_status_checks_per_second
+
+        super().__init__(
+            workflow,
+            dag,
+            cores,
+            jobname=jobname,
+            printreason=printreason,
+            quiet=quiet,
+            printshellcmds=printshellcmds,
+            latency_wait=latency_wait,
+            cluster_config=cluster_config,
+            restart_times=restart_times,
+            assume_shared_fs=True,
+            max_status_checks_per_second=max_status_checks_per_second,
+        )
+        self.exec_job += " --slurm-jobstep"
+        print(self.exec_job)
+
+    def shutdown(self):
+        """
+        waiting for all jobs to be ready
+        """
+        super().shutdown()
+        # perform additional steps on shutdown if necessary
+
+    def cancel(self):
+        for job in self.active_jobs:
+            jobid = job.jobid
+            try:
+                # timeout set to 60, because a scheduler cycle usually is
+                # about 30 sec, but can be longer in extreme cases.
+                # Under 'normal' circumstances, 'scancel' is executed in
+                # virtually no time.
+                subprocess.run(shlex.split("scancel {}".format(jobid)), timeout=60)
+            except subprocess.TimeoutExpired:
+                pass  # shell we ignore the timeout, here?
+        self.shutdown()
+
+    def run_jobs(self, jobs, callback=None, submit_callback=None, error_callback=None):
+        """Run a list of jobs that is ready at a given point in time.
+
+        By default, this method just runs each job individually.
+        This behavior is inherited and therefore this method can be removed from the skeleton if the
+        default behavior is intended.
+        This method can be overwritten to submit many jobs in a more efficient way than one-by-one.
+
+        Note that in any case, for each job, the callback functions have to be called individually!
+        """
+        # if groupjob:
+        #    attributes of groubjobs (snakemake/jobs.py class GroupJob , _resources
+        #    - see topology
+
+        for job in jobs:
+            # TODO replace by own code
+            self.run(
+                job,
+                callback=callback,  # to be executed upon ready jobs
+                submit_callback=submit_callback,  # to be hold for --immediate flag - ignored within this executor
+                error_callback=error_callback,  #
+            )
+
+    @property
+    def tmpdir(self):
+        if self._tmpdir is None:
+            self._tmpdir = tempfile.mkdtemp(dir=".snakemake", prefix="tmp.")
+        return os.path.abspath(self._tmpdir)
```

Comment:
```suggestion
```

inherit

Reply:
Not sure what files you mean, you'll have to explain further. With the request here I just meant to say that this method is present in the superclass and hence does not need to be redefined here.